include/fiction/technology/charge_distribution_surface.hpp

```diff
@@ -40,75 +45,720 @@ class charge_distribution_surface<Lyt, false> : public Lyt
+     * @param input2 The second input index.
+     * @return The distance index between index1 and index2 (indices correspond to unique SiDBs).
+     */
+    [[nodiscard]] constexpr double get_distance_index(const uint64_t& input1, const uint64_t& input2)
+    {
+        return strg->dist_mat[input1][input2];
+    }
+
+    /**
+     * Calculates and returns the electrostatic potential between two cells.
+     *
+     * @param input1 The first cell
+     * @param input2 The second cell
+     * @return The potential between input1 and input2
+     */
+    [[nodiscard]] constexpr double get_potential_cell(const typename Lyt::cell& cell1, const typename Lyt::cell& cell2)
+    {
+        auto index1 = cell_to_index(cell1);
+        if (auto index2 = cell_to_index(cell2); (index1 != -1) && (index2 != -1))
+        {
+            return strg->pot_mat[static_cast<uint64_t>(index1)][static_cast<uint64_t>(index2)];
+        }
+        return 0;
+    }
+
+    /**
+     * Calculates and returns the potential of two indices.
+     *
+     * @param input1 The first index
+     * @param input2 The second index
+     * @return The potential between input1 and input2
+     */
+    [[nodiscard]] constexpr double get_potential_index(const uint64_t& input1, const uint64_t& input2)
+    {
+        return strg->pot_mat[input1][input2];
+    }
+
+    /**
+     * The electrostatic potential between two cells (SiDBs) is calculated.
+     *
+     * @param input1 The first index.
+     * @param input2 The second index.
+     * @return The potential between input1 and input2.
+     */
+    [[nodiscard]] double potential_sidb_pair_index(const uint64_t& index1, const uint64_t& index2) const
+    {
+        if (strg->dist_mat[index1][index2] == 0)
+        {
+            return 0.0;
+        }
+
+        return (strg->phys_params.k / strg->dist_mat[index1][index2] *
+                std::exp(-strg->dist_mat[index1][index2] / strg->phys_params.lambda_tf) *
+                physical_sim_constants::ELECTRIC_CHARGE);
+    }
+
+    /**
+     * Calculates and returns the potential of a pair of cells based on their distance and simulation parameters.
+     *
+     * @param c1 The first cell.
+     * @param c2 The second cell.
+     * @return The potential between c1 and c2.
+     */
+    [[nodiscard]] double potential_sidb_pair_cell(const typename Lyt::cell& c1, const typename Lyt::cell& c2) const
+    {
+        auto index1 = cell_to_index(c1);
+        auto index2 = cell_to_index(c2);
+
+        if (strg->dist_mat[index1][index2] == 0)
+        {
+            return 0.0;
+        }
+        return (strg->sim_params.k / strg->dist_mat[index1][index2] *
+                std::exp(-strg->dist_mat[index1][index2] / strg->sim_params.lambda_tf) *
+                physical_sim_constants::ELECTRIC_CHARGE);
+    }
+
+    /**
+     * The function calculates the electrostatic potential for each SiDB position (local).
+     */
+    void local_potential()
```

Comment:
Name indicates it should return something. `initialize_local_potential`? Should it be private?